services/consensuscontext/test/consensus_context_test.go

```diff
@@ -0,0 +1,70 @@
+package test
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+/*
+
+Questions:
+
+1. What is weighted random sorting algo, and do we use reputation here
+2. "minimal-block-delay-sec" - max wait time for tx? so should be called "max..."
+3. metadata placeholder
+4.
+
+
+*/
+
+func TestRequestOrderingCommittee(t *testing.T) {
+	h := newHarness()
+	blockHeight := primitives.BlockHeight(1)
+	federationSize := len(h.config.FederationNodes(uint64(blockHeight)))
+
+	t.Run("if committee size <= federation size, then return same count of committee members as requested by config", func(t *testing.T) {
```

Comment:
why do you use this nesting? you're only sharing the harness and two simple variables between tests

Reply:
I find this hierarchical construct clearer than the flat one, where harness init is repeated every test. Also, shared data between tests and overall number of tests in this suite will likely increase.